app/javascript/packages/validated-field/validated-field-element.ts

```diff
@@ -117,20 +119,38 @@ class ValidatedFieldElement extends HTMLElement {
    */
   getOrCreateErrorMessageElement(): Element {
     if (!this.errorMessage) {
-      this.errorMessage = this.ownerDocument.createElement('div');
-      this.errorMessage.classList.add('usa-error-message');
-      if (this.descriptorId) {
-        this.errorMessage.id = this.descriptorId;
+      this.errorMessage =
+        (this.descriptorId && this.ownerDocument.getElementById(this.descriptorId)) ||
```

Comment:
I was trying to understand why we needed this changes. It looks like the approach Shannon and I had initially taken with updating to get the existing error message by `descriptorId` wasn't working quite right, since the element is described by _two_ elements, so `getElementById` would fail.

```
this.descriptorId
// 'validated-field-error-e8c668a7 validated-field-hint-e8c668a7'
```

Maybe we could check each of those as candidate error message elements?

```ts
  getDescribedErrorElement(): HTMLElement | null {
    const describedByElements = this.descriptorId?.split(' ').map(document.getElementById) || [];
    for (const element of describedByElements) {
      if (element?.classList.contains('usa-error-message')) {
        return element;
      }
    }

    return null;
  }
```

Then we can remove the changes here and update `connectedCallback` with:

```ts
this.errorMessage = this.getDescribedErrorElement() || this.querySelector('.usa-error-message');
```

Reply:
I'll take a look at this early next week.

Reply:
I implemented a very similar function, except I 1) made it `private`, 2) used `find` instead of the loop, and 3) included the `querySelector` call in the function.

I also added tests to verify that the functionality works as expected, including when the element has multiple IDs in `aria-describedby`.

Reply:
Do we still need the revisions to this `getOrCreateErrorMessageElement` function now that we've fixed up the `this.errorMessage` assignment in `connectedCallback`, or can we revert it back to what it was originally?